src/Serialization/Serializer.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Xml.Serialization;
+using System.Xml;
+using System.IO;
+
+using ShadowsOfShadows.Entities;
+using ShadowsOfShadows.Items;
+
+namespace ShadowsOfShadows.Serialization
+{
+    public static class Serializer
+    {
+        public static void Save(SaveSlot slot, GameState state)
+        {
+            // TODO Catch exception
+            XmlSerializer xsSubmit = new XmlSerializer(typeof(GameState),
+                new Type[] { typeof(RegenerationConsumable), typeof(Apple),
+                    typeof(Wall), typeof(Chest)}); // TODO Is it necessary to list all the types here?
+            var xml = "";
+
+            using (var sww = new StringWriter())
+            {
+                using (XmlWriter writer = XmlWriter.Create(sww))
+                {
+                    xsSubmit.Serialize(writer, state);
+                    xml = sww.ToString();
+
+                    System.IO.StreamWriter file = new System.IO.StreamWriter("../../../savedgames/" + slot + ".sav");
```

Comment:
Czemu idziemy `../../../`? W momencie kiedy gra opuści nasz folder `bin/Debug` to będziemy chcieli żeby save'y siedziały np. w folderze domowym użytkownika.

Wyciągnij ten początkowy string do pola `private static string SaveFolder` i można zrobić 
```
private static string SaveFolder = Path.Combine(
                        Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData)),
                        "ShadowsOfShadows",
                        "savedgames");
```